drivers/sensors/cirque_pinnacle.c

```diff
@@ -207,30 +221,51 @@ void cirque_pinnacle_init(void) {
 
     touchpad_init = true;
 
-    // Host clears SW_CC flag
-    cirque_pinnacle_clear_flags();
-
     // send a RESET command now, in case QMK had a soft-reset without a power cycle
     RAP_Write(HOSTREG__SYSCONFIG1, HOSTREG__SYSCONFIG1__RESET);
     wait_ms(30); // Pinnacle needs 10-15ms to boot, so wait long enough before configuring
     RAP_Write(HOSTREG__SYSCONFIG1, HOSTREG__SYSCONFIG1_DEFVAL);
     wait_us(50);
 
-    // FeedConfig2 (Feature flags for Relative Mode Only)
+    // Host clears SW_CC flag
+    cirque_pinnacle_clear_flags();
+
+#if CIRQUE_PINNACLE_POSITION_MODE
     RAP_Write(HOSTREG__FEEDCONFIG2, HOSTREG__FEEDCONFIG2_DEFVAL);
+#else
+    // FeedConfig2 (Feature flags for Relative Mode Only)
+    uint8_t feedconfig2 = HOSTREG__FEEDCONFIG2__GLIDE_EXTEND_DISABLE | HOSTREG__FEEDCONFIG2__INTELLIMOUSE_MODE;
+#    if !defined(CIRQUE_PINNACLE_TAP_ENABLE)
+    feedconfig2 |= HOSTREG__FEEDCONFIG2__ALL_TAP_DISABLE;
+#    endif
+#    if !defined(CIRQUE_PINNACLE_SECONDARY_TAP_ENABLE)
+    feedconfig2 |= HOSTREG__FEEDCONFIG2__SECONDARY_TAP_DISABLE;
+#    endif
+#    if !defined(CIRQUE_PINNACLE_SIDE_SCROLL_ENABLE)
+    feedconfig2 |= HOSTREG__FEEDCONFIG2__SCROLL_DISABLE;
+#    endif
+    RAP_Write(HOSTREG__FEEDCONFIG2, feedconfig2);
+#endif
 
     // FeedConfig1 (Data Output Flags)
     RAP_Write(HOSTREG__FEEDCONFIG1, CIRQUE_PINNACLE_POSITION_MODE ? HOSTREG__FEEDCONFIG1__DATA_TYPE__REL0_ABS1 : HOSTREG__FEEDCONFIG1_DEFVAL);
 
+#if CIRQUE_PINNACLE_POSITION_MODE
     // Host sets z-idle packet count to 5 (default is 0x1E/30)
     RAP_Write(HOSTREG__ZIDLE, 5);
+#endif
 
+#if CIRQUE_PINNACLE_ATTENUATION != EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_4X
     cirque_pinnacle_set_adc_attenuation(CIRQUE_PINNACLE_ATTENUATION);
+#endif
+
 #ifdef CIRQUE_PINNACLE_CURVED_OVERLAY
     cirque_pinnacle_tune_edge_sensitivity();
 #endif
+#if (CIRQUE_PINNACLE_ATTENUATION != EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_4X) || defined(CIRQUE_PINNACLE_CURVED_OVERLAY)
     // Force a calibration after setting ADC attenuation
     cirque_pinnacle_calibrate();
+#endif
```

Comment:
I had read out the default value for ADC attenuation on a freshly powered up TM040040 and it was 2x (0x4E in EXTREG__TRACK_ADCCONFIG). Do you see any harm in recalibrating anyway?

Reply:
Calibration adds 100-200 ms delay. I'll change that to read the register instead of ifdef